website/docs/examples/record_track/README.md

```diff
@@ -0,0 +1,163 @@
+                        The name of the track.
+  --output-dir OUTPUT_DIR
+                        The directory to save the track to.
+```
+
+### 5. Code overview
+
+In this example we use the `EventClient` with the `subscribe` method to receive the filter track stream.
+
+In this example, we:
+
+- Request that the filter reset the tracking of the filter track with the `/clear_track` request
+  - This will clear any previously hit criteria that would cause the filter to stop adding to the
+    track. See above for more details.
+- Stream the `/track` topic from the filter service
+- Write (or overwrite) the full `FilterTrack` proto to disk every time a pose is added to the track
+  - This allows you to exit the program at anytime with the track you have recorded up to that point
+
+```python
+async def main(service_config_path: Path, track_name: str, output_dir: Path) -> None:
+    """Run the filter service client to record a track.
+
+    Args:
+        service_config_path (Path): The path to the filter service config.
+        track_name (str): The name of the track.
+        output_dir (Path): The directory to save the track to.
+    """
+    # create a client to the filter service
+    config: EventServiceConfig = proto_from_json_file(service_config_path, EventServiceConfig())
+
+    # Clear the track so everything going forward is tracked
+    await EventClient(config).request_reply("/clear_track", Empty())
+
+    # Create a list to store the filter track states
+    filter_track_states: list[FilterState] = []
+
+    # Subscribe to the filter track topic
+    async for event, message in EventClient(config).subscribe(config.subscriptions[0], decode=True):
+
+        print("###################")
+        print("Adding to track:")
+        print(message)
+
+        # Add the filter state to the list
+        filter_track_states.append(message)
+
+        # Write the FilterTrack to disk, overwriting the file each time
+        if not proto_to_json_file(
+            output_dir / f"{track_name}.json", FilterTrack(states=filter_track_states, name=track_name)
+        ):
+            raise RuntimeError(f"Failed to write track to {output_dir}")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(prog="amiga-record-track")
+    parser.add_argument("--service-config", type=Path, required=True, help="The filter service config.")
+    parser.add_argument("--track-name", type=str, required=True, help="The name of the track.")
+    parser.add_argument(
+        "--output-dir", type=Path, default=Path(__file__).parent,
+        help="The directory to save the track to."
+    )
+    args = parser.parse_args()
+
+    if not args.output_dir.exists() or not args.output_dir.is_dir():
+        raise ValueError(f"Invalid output directory: {args.output_dir}")
+
+    if not args.track_name:
+        raise ValueError("No track name provided.")
+
+    asyncio.run(main(args.service_config, args.track_name, args.output_dir))
+```
+
```

Comment:
This may sound dumb, but I would give them direct instructions to take the pendant and start driving the Amiga. Alternatively, you can also add instructions to teach users how to use WASD keys to command the Amiga and record the track (that's how I tested it, and I thought it was pretty cool).

Reply:
Good idea!